File: src/main/java/org/goodgoodgood/freetubeexservice/controller/MusicController.java
package org.goodgoodgood.freetubeexservice.controller;

import lombok.RequiredArgsConstructor;
import org.goodgoodgood.freetubeexservice.config.auth.SessionUser;
import org.goodgoodgood.freetubeexservice.domain.music.MusicService;
import org.goodgoodgood.freetubeexservice.dto.music.AddMusicDto;
import org.goodgoodgood.freetubeexservice.dto.music.CreatePlaylistDto;
import org.goodgoodgood.freetubeexservice.dto.music.PlaylistDto;
import org.goodgoodgood.freetubeexservice.dto.music.PlaylistsShortDto;
import org.springframework.web.bind.annotation.*;

import javax.servlet.http.HttpSession;
import java.util.List;

@RequiredArgsConstructor
@RestController
public class MusicController {

    private final HttpSession httpSession;
    private final MusicService musicService;

    @GetMapping("api/playlists")
    public List<PlaylistsShortDto> getPlaylists() {
        SessionUser user = (SessionUser) httpSession.getAttribute("user");

        return musicService.getPlaylistsByEmail(user.getEmail());
    }

    @GetMapping("api/playlists/{playlistId}")
    public PlaylistDto getPlaylistDetail(@PathVariable Long playlistId) {
        SessionUser user = (SessionUser) httpSession.getAttribute("user");

        return musicService.getPlaylist(user.getEmail(), playlistId);
    }

    @PostMapping("api/playlists")
    public Long createPlaylist(@RequestBody CreatePlaylistDto dto) throws Exception {
        SessionUser user = (SessionUser) httpSession.getAttribute("user");

        return musicService.createPlaylist(user.getEmail(), dto);
    }

    @PostMapping("api/playlists/{playlistId}/musics")
    public Long addMusic(
            @PathVariable Long playlistId,
            @RequestBody AddMusicDto dto
    ) {
        SessionUser user = (SessionUser) httpSession.getAttribute("user");

        return musicService.addMusic(playlistId, user.getEmail(), dto);
    }

}

File: src/main/kotlin/org/goodgoodgood/freetubeexservice/controller/MusicController.kt
package org.goodgoodgood.freetubeexservice.controller

import org.goodgoodgood.freetubeexservice.config.auth.SessionUser
import org.goodgoodgood.freetubeexservice.domain.music.MusicService
import org.goodgoodgood.freetubeexservice.domain.music.dto.AddMusicDto
import org.goodgoodgood.freetubeexservice.domain.music.dto.CreatePlaylistDto
import org.goodgoodgood.freetubeexservice.domain.music.dto.PlaylistDto
import org.goodgoodgood.freetubeexservice.domain.music.dto.PlaylistsShortDto
import org.springframework.web.bind.annotation.*
import javax.servlet.http.HttpSession

@RestController
class MusicController(
    private val httpSession: HttpSession,
    private val musicService: MusicService,
) {

    @GetMapping("api/playlists")
    fun getPlaylists(): List<PlaylistsShortDto> {
        val user = httpSession.getAttribute("user") as SessionUser
        return musicService.getPlaylistsByEmail(user.email)
    }

    @GetMapping("api/playlists/{playlistId}")
    fun getPlaylistDetail(@PathVariable playlistId: Long): PlaylistDto {
        val user = httpSession.getAttribute("user") as SessionUser
        return musicService.getPlaylist(user.email, playlistId)
    }

    @PostMapping("api/playlists")
    fun createPlaylist(@RequestBody dto: CreatePlaylistDto): Long {
        val user = httpSession.getAttribute("user") as SessionUser
        return musicService.createPlaylist(user.email, dto)
    }

    @PostMapping("api/playlists/{playlistId}/musics")
    fun addMusic(
        @PathVariable playlistId: Long,
        @RequestBody dto: AddMusicDto,
    ): Long {
        val user = httpSession.getAttribute("user") as SessionUser
        return musicService.addMusic(playlistId, user.email, dto)
    }

}